# Robot namespace
robot_name: mobile_manipulator

# Name of manipulator joints in URDF model in order from waist to wrist
manipulator_dof_names: 
  - waist_joint
  - shoulder_joint
  - elbow_joint
  - upper_forearm_joint
  - lower_forearm_joint
  - wrist_joint

# Manipulator Controllers in order from waist to wrist
manipulator_controllers: 
  - waist_controller
  - shoulder_controller
  - elbow_controller
  - upper_forearm_controller
  - lower_forearm_controller
  - wrist_controller

# Parameter to select the way to calculate the orientation error
# 1 -> Angle-axis Munoz-Osorio
# 2 -> Angle-axis Caccavale
# 3 -> Quaternion Yuan
# 4 -> Quaternion Caccavale 1
# 5 -> Quaternion Caccavale 2
orientation_error_calc: 5

# Parameter to define if only using manipulator (For testing purposes)
only_manipulator: True

# Parameters of controller
compensate_topdownEffects: False # True False
compensate_nonlinearInJointSpace: True
use_augmented_projections: True

# Maximum velocities when using the controllers: Go to point with constant vel
point_const_vel_max_lineal:  0.3
point_const_vel_max_angular: 3.14

# Controller gains

Cartesian_position_gains:
  stiffness:
    X: 400.0
    Y: 400.0
    Z: 400.0
  damping_ratios:
    X: 0.9
    Y: 0.9
    Z: 0.9

Cartesian_orientation_gains:
  stiffness:
    X: 400.0
    Y: 400.0
    Z: 400.0
  damping_ratios:
    X: 0.9
    Y: 0.9
    Z: 0.9

# Parameter for singularity handling
# 0 -> Algorithm proposed by Khatib et al.
# 1 -> Algorithm without torque projection
# 2 -> Algorithm with non singular torque projection
singularity_handling: 1

# Parameter for joint limit avoidance
# -1 -> Not handling joint limits
#  0 -> Repulsive artificial potentials
#  1 -> Intermediate value
#  3 -> SJS
joint_limit_avoidance: 0